skimage/io/tests/test_imread.py

```diff
@@ -12,16 +12,16 @@
     fetch,
 )
 
-from pytest import importorskip
+import pytest
 
-importorskip('imread')
+pytest.importorskip('imread')
 
 
-def setup():
+@pytest.fixture(autouse=True)
+def _use_imread_plugin():
```

Comment:
Disclaimer: non-expert here :wink: Just curious, from reading the [documentation you linked to](https://docs.pytest.org/en/latest/deprecations.html#nose-deprecation), I was expecting some `def setup()` to be replaced by `def setup_method()` (while the fixture is recommended for replacing code using `@with_setup`)...(?)

Reply:
Ah, I didn't know that pytest supports another approach to achieve this. :sweat_smile: Though, in this case it should be [`setup_module` and `teardown_module`](https://docs.pytest.org/en/latest/how-to/xunit_setup.html#module-level-setup-teardown).

Looks like both approaches are valid but I feel like the fixture based one may the better known approach and is also the one recommended by pytest. And that I feel more comfortable with the fixture based approach may also be a factor. ^^